### PR TITLE
[search] Prefer prevEmit results in NearbyPointsSweeper

### PR DESCRIPTION
**Pull request by tatiana-yan**

### CODE REVIEW

**Comment by tatiana-yan** [comment]
JTTP

File: geometry/nearby_points_sweeper.hpp [hunk 1]
```diff
@@ -28,7 +29,7 @@ class NearbyPointsSweeper
   // Adds a new point (|x|, |y|) on the plane. |index| is used to
   // identify individual points, and will be reported for survived
   // points during the Sweep phase.
-  void Add(double x, double y, size_t index);
+  void Add(double x, double y, size_t index, uint8_t priority);
```
Comment (mpimenov) [inline geometry/nearby_points_sweeper.hpp]: Please update the documentation

**Comment by tatiana-yan** [comment]
>what was wrong with emplace?

It does not work with aggregate initialization.
I can add constructor for LineEvent but the plan was to make it as simple as possible.

File: geometry/nearby_points_sweeper.hpp [hunk 2]
```diff
@@ -47,11 +48,12 @@ class NearbyPointsSweeper
       {
         if (line.empty())
         {
-          line.emplace(event.m_x, event.m_index);
+          line.insert({event.m_x, event.m_index, event.m_priority});
           break;
         }
 
-        auto it = line.upper_bound(std::make_pair(event.m_x, std::numeric_limits<size_t>::max()));
+        auto it = line.upper_bound(
```
[the remark this answers is not in the document]
Comment (tatiana-yan) [reply geometry/nearby_points_sweeper.hpp]: тут разные типы: size_t и  uint8_t

**Comment by tatiana-yan** [comment]
@mpimenov PTAL

File: geometry/nearby_points_sweeper.hpp [hunk 3]
```diff
@@ -28,7 +29,8 @@ class NearbyPointsSweeper
   // Adds a new point (|x|, |y|) on the plane. |index| is used to
   // identify individual points, and will be reported for survived
   // points during the Sweep phase.
-  void Add(double x, double y, size_t index);
+  // Points with higher |priority| will be prefered at the Sweep phase.
```
Comment (mpimenov) [inline geometry/nearby_points_sweeper.hpp]: s/prefered/preferred/
s/at the Sweep phase/during the Sweep phase/

File: geometry/nearby_points_sweeper.hpp [hunk 4]
```diff
@@ -65,11 +68,16 @@ class NearbyPointsSweeper
...
           if (fabs(x - event.m_x) <= m_eps)
           {
-            add = false;
-            break;
+            if (it->m_priority >= event.m_priority)
+            {
+              add = false;
+              break;
+            }
+            // New event has higher priority than existing event nearby.
```
Comment (mpimenov) [inline geometry/nearby_points_sweeper.hpp]: s/existing event/an existing event/

**Comment by mpimenov** [comment]
Tests?

**Comment by tatiana-yan** [comment]
JTTP

File: geometry/nearby_points_sweeper.hpp (hunk 3 above)
Comment (tatiana-yan) [reply geometry/nearby_points_sweeper.hpp]: done

File: geometry/nearby_points_sweeper.hpp (hunk 4 above)
Comment (tatiana-yan) [reply geometry/nearby_points_sweeper.hpp]: done

**Comment by tatiana-yan** [comment]
>Tests?

done

**Pull request merged**